pkg/agent/datamodel/types.go

```diff
@@ -1079,6 +1114,14 @@ func (l *LinuxProfile) HasCustomNodesDNS() bool {
 	return false
 }
 
+// IsCSIProxyEnabled returns true if csi proxy service should be enable for Windows nodes
+func (w *WindowsProfile) IsCSIProxyEnabled() bool {
+	if w.EnableCSIProxy != nil {
+		return *w.EnableCSIProxy
+	}
+	return DefaultEnableCSIProxyWindows
+}
```

Comment:
can we have this defaults in RP and have RP pass in the final value?

Reply:
Good idea, thanks, so that RP can take control over the default behavior, we can do this in the next PR after we have the default value in PR.

Reply:
ok. looking forward to the next PR